mock/world/builtinFunctionsMock.go

```diff
@@ -79,15 +79,58 @@ func (b *MockWorld) runESDTTransferCall(input *vmcommon.ContractCallInput) (*vmi
 		return nil, ErrInsufficientFunds
 	}
 
-	return &vmcommon.VMOutput{
+	vmOutput := &vmcommon.VMOutput{
 		ReturnData:      make([][]byte, 0),
 		ReturnCode:      vmcommon.Ok,
 		ReturnMessage:   "",
-		GasRemaining:    input.GasProvided - input.GasLocked,
+		GasRemaining:    vmInput.GasProvided - vmInput.GasLocked,
 		GasRefund:       big.NewInt(0),
 		OutputAccounts:  make(map[string]*vmcommon.OutputAccount),
 		DeletedAccounts: make([][]byte, 0),
 		TouchedAccounts: make([][]byte, 0),
 		Logs:            make([]*vmcommon.LogEntry, 0),
-	}, nil
+	}
+
+	isSCCallAfter := len(vmInput.Arguments) > 2 && b.IsSmartContract(vmInput.RecipientAddr)
+
+	if isSCCallAfter {
+		var callArgs [][]byte
+		if len(vmInput.Arguments) > 3 {
+			callArgs = vmInput.Arguments[3:]
+		}
+		addOutputTransferToVMOutput(
+			string(vmInput.Arguments[2]),
```

Comment:
Maybe have this as functionName := string(vmInput.Arguments[2]) right before the second if, so it's a bit more clear what it signifies